lib/plugins/aws/provider/awsProvider.test.js

```diff
@@ -112,7 +112,7 @@ describe('AwsProvider', () => {
 
   describe('#request()', () => {
     beforeEach(() => {
-      sinon.stub(global, 'setTimeout', (cb) => { cb(); });
+      sinon.stub(global, 'setTimeout').callsFake((cb) => { cb(); });
```

Comment:
The timeout stub should be reset again after the tests? Otherwise the stubbed version will be used by all subsequent tests without knowing.

Reply:
Nice catch! Thanks @HyperBrain 👍

Reply:
Just a small improvement: Instead of faking a standard cb function you can just use `sinon.stub(global, 'setTimeout').yields()` instead. "yields" is similar to "returns" or "resolves", but just for callback based functions. You can specify error and data as parameters (e.g. `yields(new Error('myerror'))` will trigger a callback with an error and `yields(null, myData)` will trigger the callback with the given data.
Just use yields whenever you stub a function that has a callback as the last parameter. Makes things much easier.

Reply:
Good insights 👍. Thanks for that @HyperBrain! Yes, `yielding` is a little bit uncommon since most of the stuff is Promise based nowadays.